src/sm/sm_main/debug/full-tracer.js

```diff
@@ -362,8 +362,18 @@ class FullTracer {
         // Append logs correctly formatted to response logs
         this.logs = this.logs.filter((n) => n); // Remove null values
         // eslint-disable-next-line no-restricted-syntax
-        for (const l of this.logs) {
-            this.finalTrace.responses[this.txCount].logs = this.finalTrace.responses[this.txCount].logs.concat(Object.values(l));
+        let logIndex = 0;
+        for (let i = 0; i < this.logs.length; i++) {
+            const logsToAdd = Object.values(this.logs[i]);
```

Comment:
all the logs on CTX 1 should not be before ALL the logs in CTX2.
Should be indexed the same order they were emitted